gravatar-ui/src/main/java/com/gravatar/ui/components/atomic/Avatar.kt

```diff
@@ -88,16 +91,24 @@ public fun Avatar(
  * @param state Avatar URL wrapped in ComponentState
  * @param size The size of the avatar
  * @param modifier Composable modifier
+ * @param avatarQueryOptions Options to customize the avatar query
  */
 @JvmName("AvatarWithURLComponentState")
 @Composable
-public fun Avatar(state: ComponentState<String>, size: Dp, modifier: Modifier = Modifier) {
+public fun Avatar(
+    state: ComponentState<URI>,
+    size: Dp,
+    modifier: Modifier = Modifier,
+    avatarQueryOptions: AvatarQueryOptions? = null,
+) {
     when (state) {
         is ComponentState.Loading -> SkeletonAvatar(size = size, modifier = modifier)
 
         is ComponentState.Loaded -> {
             Avatar(
-                model = state.loadedValue,
+                model = state.loadedValue.toURL()?.let { url ->
+                    URL(url.protocol, url.host, url.path.plus(avatarQueryOptions.asQueryParameters()))
+                }.toString(),
```

Comment:
Couldn't we use the `AvatarUrl` like we do in the other `Avatar` composable that takes the full profile?

Reply:
In the `ProfileCard` the avatar URI after the selection is different `https://www.gravatar.com/userimage/...`. I wonder if this is okey to use the params that are defined for `https://gravatar.com/avatar/HASH`

Maybe instead of passing the `/userimage` we should pass the `/avatar` with the cache buster... 🤔

Reply:
> Couldn't we use the AvatarUrl like we do in the other Avatar composable that takes the full profile?

`/userimage` doesn't contain the user email hash, and `AvatarUrl` works by using that hash so we couldn't use it in this case.

> Maybe instead of passing the /userimage we should pass the /avatar with the cache buster... 🤔

Yeah, it sounds like a valid approach. What do you think about adding a `cacheBuster` parameter to `AvatarQueryOptions` ? That way, we'll open it also to third-party developers.

Reply:
I believe the `AvatarQueryOptions` were designed to reflect [this](https://docs.gravatar.com/api/avatars/images/). At least the public stuff. We could have it there as internal though.

Reply:
We can see the `cacheBuster` as part of the public queryParams. Basically, we need it in some cases, for example, in the QE, but any third-party would require the same if they have a custom Avatar component but use our SDK to upload and generate the AvatarUrls. (That can be us in PocketCast, for example).

> We could have it there as internal though.

I don't think it can be internal because it's used from the QE.

Reply:
>I don't think it can be internal because it's used from the QE.

Right.

I would avoid adding it as a public API because we might be able to change it if the server starts respecting the no-cache header. Once we add it to the public API we won't be able to easily remove it. 

What about following iOS:
- Add a `forceRefresh` param to `Avatar` composable
- Append the cache buster query separately

So in general, I would add an option to force a fresh image to our SDK so that third-party apps can use i, but keep the implementation internal. 

WDYT?

Reply:
> I would avoid adding it as a public API because we might be able to change it if the server starts respecting the no-cache header.

I don't know if that would be anytime soon, but it is a fair point.

About the `forceRefresh` approach, I have some doubts, but let me think and play a bit more with the code.

Reply:
I am still considering an approach to `forceRefresh` that matches well with Jetpack Compose. I'll try to come up with something tomorrow, but if you have some ideas, feel free to share them.

Reply:
I've opened #350 to discuss the `forceRefresh` approach. If we decide to move forward with that, I'll close this one.